### ISSUE TITLE
Outstrument issue for Malli's multi-arity schema with `m/=>`

### ISSUE THREAD

**Issue by rynkowsg** [CrypticButter/snoop#1]
There is an issue with outstrumenting mutli-arity fn when the schema is provided with `m/=>`:

```
(m/=> print [:function
             [:=> [:cat string?] nil?]
             [:=> [:cat string? string?] nil?]])

(>defn print
  ([first] (println first))
  ([first second] (println first second)))

(print "Hello!")
; => Hello!
;    Outstrument error for: print
;    Got: nil
;    Error: ["invalid function"]
;    Execution error (ExceptionInfo) at crypticbutter.snoop.config/throw-validation-error (config.cljc:42).
;    Outstrument failed. See message printed above.
```

- Malli docs: https://github.com/metosin/malli#clj-kondo
- malli tests involving `:function`: https://github.com/metosin/malli/blob/fc4b04207f958723c37c51dcc688d2c13217caff/test/malli/core_test.cljc#L2139

**Comment by LuisThiamNye** [comment]
Indeed, `snoop` does not currently support reading schemas using the `:function` syntax.

The support could be implemented, but malli now has its own [built-in instrumentation features](https://github.com/metosin/malli/blob/master/docs/function-schemas.md#instrumentation). Considering that you are using `m/=>`, you might find that a better fit. If not, let me know.

**Comment by rynkowsg** [comment]
TBH I like Snoop. The thing is that I couldn't find a way of sharing schemas for the multi-arity functions in a concise way. That's why I give a try with` m/=>` as it looked it is supported.

In my code, I have got several multi-arity functions sharing exactly the same in/out interface. How would you share a schema among them?

**Comment by LuisThiamNye** [comment]
Right I understand the problem, I will work on that.

And out of curiosity, what is it about `snoop` that you prefer over the other solutions? With the release of malli.instrument, I have to figure out what unique value it can provide in order to avoid unnecessarily reimplementing lots of functionality.

**Comment by LuisThiamNye** [comment]
I've made some changes in "21-201-alpha" — see if that works

**Comment by rynkowsg** [comment]
It works now. Thanks.

I think Snoop is the best tool available now to provide an Orchestra-like experience, if not better. I used to work with Orchestra before. When experimenting with Malli I checked aave and malli-instrument at first, but the former was not obvious how to integrate, and with the latter I had issues when playing with REPL. 

Snoop, on the other hand, just works. Easy to integrate. No issues working heavily with schemas in REPL. And the macros are quite elastic. At first, it looked weird that so many syntaxes for providing schema are possible, usually libraries impose some style, but in the end, I like that this choice is given. I experimented with three different syntaxes you provide and I've found what looks/works best for me (it was neither first nor second choice).

In overall, great tool! Thanks for making it ;)
